core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1974,9 +1973,10 @@ protected Type visitTrim(Trim node, StackableAstVisitorContext<Context> context)
             argumentTypes.add(process(node.getTrimSource(), context));
             arguments.add(node.getTrimSource());
             node.getTrimCharacter().ifPresent(trimChar -> {
-                process(trimChar, context);
+                Type trimCharType = process(trimChar, context);
+                coerceType(trimChar, trimCharType, CODE_POINTS, "Function trim argument 2");
```

Comment:
You don't need to pass the `CODE_POINTS` type to the function resolver. It will take care of finding the right coercion. The right order of events would be to:
1. process both arguments: trimSource and trimChar
2. resolve function using the _actual_ argument types
3. check the function signature for the _expected_ argument types
4. apply coercions (the `coerceType()` method) to both arguments